### PR TITLE
Update vagrant setup to Fedora 38

### PR DESCRIPTION
**Pull request by mpeters**
This also includes lots of other fixes/improvements:
    
- Pull in ansible-keylime role from github instead of including parts of it ourselves
- Install rust environment and keylime rust agent repo (if configured)
- Install swtpm from package instead of source
- Install ima-emulator and tpm-emulator systemd files that used to be part of keylime ima_stub
- Improved ansible verbose handling
- Configure IMA config directly
- Reboot after changing IMA config
- Remove lots of unnecessary bits

### CODE REVIEW

File: roles/tpm_setup/files/ima_emulator.service [hunk 1]
```diff
@@ -0,0 +1,11 @@
+[Unit]
+Description=IMA emulator for Keylime
+After=network.target tpm_emulator.service
+Requires=tpm_emulator.service
+
+[Service]
+Environment=TPM2TOOLS_TCTI="swtpm:port=2321"
+ExecStart=/usr/bin/bash -c "keylime_ima_emulator"
```
[the remark this answers is not in the document]
Comment (mpeters) [reply roles/tpm_setup/files/ima_emulator.service]: Good idea. Done

**Comment by THS-on** [comment]
Have you considered using the vtpm_proxy Kernel feature (https://github.com/stefanberger/swtpm/wiki/Using-the-Intel-TSS-with-swtpm#character-device-using-tpm_vtpm_proxy)? This has the advantage that the vTPM looks like a normal TPM under /dev/tpm{,rm}0

**Comment by mpeters** [comment]
> Have you considered using the vtpm_proxy Kernel feature (https://github.com/stefanberger/swtpm/wiki/Using-the-Intel-TSS-with-swtpm#character-device-using-tpm_vtpm_proxy)? This has the advantage that the vTPM looks like a normal TPM under /dev/tpm{,rm}0

I would like to explore better vtpm setups to make it less different than a real tpm. I'm also looking at possibly using the virtualization layer's tpm which would make it work with measured boot (which the vtpm_proxy wouldn't be able to do right?).

But for this initial change, I just wanted to get it back to working pretty close to what was happening before.

**Comment by THS-on** [comment]
> which the vtpm_proxy wouldn't be able to do right?.

We could write a UEFI emulator, to do the same what we are doing with IMA or the feature in the IBM TSS parser that already can do that, but I agree with you that if we could attach a TPM at the VM layer this would be ideal.

**Review by THS-on — APPROVED** [review]

**Pull request merged**